smart-contracts/contracts/mixins/MixinTransfer.sol

```diff
@@ -85,6 +86,9 @@ contract MixinTransfer is
       _assignNewTokenId(toKey);
       iDTo = toKey.tokenId;
       _recordOwner(_to, iDTo);
+      // Assign the KeyManager
+      keyManagerOf[toKey.tokenId] = _keyManager;
```

Comment:
This is not consistent with purchase. What if the iDTo owns an expired key? We should be consistent or simplify

Reply:
the check for this block is:
`if (iDTo == 0) {...`
ensuring this only operates on new keys.